R/proxy.R

```diff
@@ -39,7 +39,7 @@ compare_proxy.default <- function(x, path) {
 #' @export
 compare_proxy.data.table <- function(x, path) {
   attr(x, ".internal.selfref") <- NULL
-  list(object = x, path = path)
```

Comment:
Do you think this is a meaningful improvement? I considered it but it seemed like it would add a lot of noise to the comparisons, without adding much insight to the user since (as I understand it) `.internal.selfref` is largely an implementation detail.

Reply:
setattr(x, '.internal.selfref', NULL) conveys it more legibly but it's not _quite_ the same.

I was thinking that if the object is to be reused in another test it might error but looking again I guess removing the attr won't bubble up to the original object in the testing environment. in which case it's fine to revert.

Reply:
In that case, I think it's slightly better as is, but I'm happy to revisit if we hit a concrete case of confusion.